components/brave_wallet/browser/zcash/zcash_transaction_utils.cc

```diff
@@ -69,27 +69,49 @@ std::optional<PickInputsResult> PickZCashTransparentInputs(
 
   if (amount == kZCashFullAmount) {
     auto total_inputs_amount = CalculateInputsAmount(all_inputs);
-    auto fee = CalculateZCashTxFee(all_inputs.size(), orchard_actions_count);
+    // Full amount case - no change output.
+    base::CheckedNumeric<uint64_t> fee = 0u;
+    switch (output_type) {
+      case ZCashTargetOutputType::kTransparent:
+        fee = CalculateZCashTxFee(all_inputs.size(), 1u, 0u, 0u);
+        break;
+      case ZCashTargetOutputType::kOrchard:
+        fee = CalculateZCashTxFee(all_inputs.size(), 0u, 0u, 1u);
+        break;
+      default:
```

Comment:
Can we have this implemented so 0s and 1s are not hardcoded but are calculated from what transaction acutally has?